plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -858,7 +863,8 @@ public void createMaterializedView(ConnectorSession session, SchemaTableName vie
                 .setParameters(viewProperties)
                 .withStorage(storage -> storage.setStorageFormat(VIEW_STORAGE_FORMAT))
                 .withStorage(storage -> storage.setLocation(""))
-                .setViewOriginalText(Optional.of(encodeMaterializedViewData(definition)))
+                .setViewOriginalText(Optional.of(encodeMaterializedViewData(
```

Comment:
I think we should extract internal and backward compatible representation of MV for iceberg and use it for serde. This way we won't break any backward compatibility.